lib/matplotlib/image.py

```diff
@@ -550,20 +550,33 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                      cbook._setattr_cm(self.norm, vmin=s_vmin, vmax=s_vmax):
                     output = self.norm(resampled_masked)
             else:
-                if A.ndim == 2:  # _interpolation_stage == 'rgba'
+                if A.ndim == 2:  # self._interpolation_stage == 'rgba'
```

Comment:
The comment would better be placed on the `else` above.

Reply:
I think it should stay with the `if` statement?
The `else` above is the logical negation of `if A.ndim == 2 and self._interpolation_stage == 'data'`, which covers `A.ndim != 2` when `self._interpolation_stage` can be either `'data'` or `'rgba'`.

Reply:
Yes, true. Sorry for the confusion.